api/src/OpenTelemetry/Logging/Core.hs

```diff
@@ -131,8 +138,87 @@ emitLogRecord Logger {..} LogRecordArguments {..} = do
       , logRecordResource = loggerProviderResource loggerProvider
       , logRecordInstrumentationScope = loggerInstrumentationScope
       , logRecordAttributes =
-          addAttributes
+          A.addAttributes
             (loggerProviderAttributeLimits loggerProvider)
-            emptyAttributes
+            A.emptyAttributes
             attributes
+      , logRecordLogger = logger
       }
+
+
+{- | Emits a LogRecord with properties specified by the passed in Logger and LogRecordArguments.
+If observedTimestamp is not set in LogRecordArguments, it will default to the current timestamp.
+If context is not specified in LogRecordArguments it will default to the current context.
+-}
+emitLogRecord
+  :: (MonadIO m)
+  => Logger
+  -> LogRecordArguments body
+  -> m (LogRecord body)
+emitLogRecord l args = do
+  ilr <- createImmutableLogRecord l args
+  lr <- liftIO $ newIORef ilr
+  pure $ LogRecord lr
```

Comment:
`emitLogRecord` now contains the `IORef` logic while `createImmutableLogRecord` contains the `LogRecord` creation logic.